### DISCUSSION TITLE
Support 2FA-required remote (and auto-login in integrated VSCode)

### DISCUSSION THREAD

**Discussion by matthewdm0816** [blinksh/blink#1730] [Ideas]
Hi,
I'm really glad to find this great tool to work on remote with the VSCode that is remote-capable with blink-fs. But as indicated in the document, the login to remote server require to fill in all authentications in advance in the "Hosts" settings. Passwords or secret keys can be provided in such way, but 2FA code (e.g. TOTP password) is not supported yet, and makes it unable to use VSCode on remote servers with 2FA verification.
Any chance to support auto 2FA generation in "Hosts" settings? Thus it'll be able to auto-login 2FA-enabled servers

**Comment by carloscabanero** [comment]
Hey, thanks for opening this. When we connect to the remote, we do so using SFTP. During the time you are using VS Code, we may need to reconnect multiple times transparently, in the background. The issue is that to reconnect with 2FA, we would need to request your permission again and again, all the time, which is not a very good experience.

One solution would be to run our own "file server" once we connect over ssh, similar to what we do with mosh. Maybe using a different protocol?

I'm moving this to Discussions in case anyone has more ideas.
